stress_test/controller.py

```diff
@@ -0,0 +1,349 @@
+#        with open(json_file) as json_conf_file:
+#        self.test_config = json.load(json_conf_file)
+
+        """Create a Controller object. Options from JSON input file
+        :param test_config: JSON input configuration
+        :type test_config: JSON configuration dictionary
+        """
+        self.__dict__ = test_config
+        self.name = test_config['controller_name']
+        logging.info('Initializing parameters for the {0} Controller '.format(self.name))
+
+        #self.host_ip = test_config['controller_host_ip']
+        self.node_ip = test_config['controller_node_ip']
+        self.node_ssh_port = test_config['controller_node_ssh_port']
+        self.node_username = test_config['controller_node_username']
+        self.node_password = test_config['controller_password']
+
+        self.logs_dir = ctrl_base_dir + test_config['controller_logs_dir']
+        self.rebuild = ctrl_base_dir + test_config['controller_rebuild']
+        self.cleanup = ctrl_base_dir + test_config['controller_cleanup']
+        self.port = test_config['controller_port']
+        self.cpu_shares = test_config['controller_cpu_shares']
+        self.logs_dir = ctrl_base_dir + test_config['controller_logs_dir']
+
+        self.build_handler = ctrl_base_dir + test_config['controller_build_handler']
+        self.start_handler = ctrl_base_dir + test_config['controller_start_handler']
+        self.stop_handler = ctrl_base_dir + test_config['controller_stop_handler']
+        self.status_handler = ctrl_base_dir + test_config['controller_status_handler']
+        self.clean_handler = ctrl_base_dir + test_config['controller_clean_handler']
+
+
+        self.java_opts = ' '.join(test_config['java_opts'])
+        self.cpid = -1
+
+        if 'controller_restconf_port' in test_config:
+           self.restconf_port = test_config['controller_restconf_port']
+           self.restconf_user = test_config['controller_restconf_user']
+           self.restconf_password = test_config['controller_restconf_password']
+
+    def cleanup(self,ssh_client=None):
+        """Wrapper to the controller cleanup handler
+
+        :param self.clean_handler: filepath to the controller cleanup handler
+        :param ssh_client : SSH client provided by paramiko to run the command
+        :type self.clean_handler: str
+        :type ssh_client: paramiko.SSHClient
+        """
+        common.command_exec_wrapper([self.clean_handler],
+                                    '[self.clean_handler]', ssh_client)
+
+    def check_controller_status(self,ssh_client=None):
+        """Wrapper to the controller status handler
+
+        :param self.status_handler: filepath to the controller status handler
+        :param ssh_client : SSH client provided by paramiko to run the command
+        :returns: '1' if controller is active, '0' otherwise
+        :rtype: str
+        :type self.status_handler: str
+        :type ssh_client: paramiko.SSHClient
+        """
+
```

Comment:
we should reuse the `command_exec_wrapper` function as above, and not repeat the low-level steps of it directly